daemon/cluster/executor/container/controller.go

```diff
@@ -305,6 +307,34 @@ func (r *controller) Wait(pctx context.Context) error {
 	return nil
 }
 
+func (r *controller) hasServiceBinding() bool {
+	if r.task == nil {
+		return false
+	}
+
+	// service is attached to a network besides the default bridge
+	for _, na := range r.task.Networks {
+		if na.Network == nil ||
+			na.Network.DriverState == nil ||
+			na.Network.DriverState.Name == "bridge" && na.Network.Spec.Annotations.Name == "bridge" {
+			continue
+		}
+		return true
+	}
+
+	// service has ingress publish port
+	// note that host port is local and it doesn't require cross node convergence
+	if r.task.Endpoint != nil {
+		for _, ePort := range r.task.Endpoint.Ports {
+			if ePort.PublishMode == api.PublishModeIngress {
+				return true
+			}
+		}
+	}
```

Comment:
Even if a service has no port published the clients could be other services on the same network. So I think its better make the shutdown graceful in all cases except the default attachment to the `bridge` network.